src/controller.c

```diff
@@ -133,17 +133,36 @@ static void __controller_exec_PIF( const void *inblock, void *outblock )
 }
 
 /**
- * @brief Probe the EEPROM interface
- *
- * Probe the EEPROM to see if it exists on this cartridge.
+ * @brief Send a block of data to the PIF and fetch the result
+ * 
+ * @param data_in PIF request data
+ * @param in_size Up to 64 bytes
+ * @param data_out PIF response data
+ * @param out_size Up to 64 bytes
+ */
+void pif_execute( const uint64_t * data_in, size_t in_size, uint64_t * data_out, size_t out_size )
+{
+    static uint64_t input[8] = { 0, 0, 0, 0, 0, 0, 0, 1 };
+    static uint64_t output[8];
+
+    assert(in_size <= sizeof(input));
+    assert(out_size <= sizeof(output));
+
+    memcpy( input, data_in, in_size );
+
+    __controller_exec_PIF( input, output );
+
+    memcpy( data_out, output, out_size );
+}
```

Comment:
Why do you need an internal `input` / `output` here? If it's just because in_size could be less than 64 bytes, that is already taken care of by `__controller_exec_PIF`, isn't it?

IOW how's this function different from `__controller_exec_PIF`?

Reply:
`pif_execute` exists for a few reasons:
1. `__controller_exec_PIF` is static and not exposed outside the translation unit.
2. `pif_execute` allows input less than 64-bytes and sets the last bit to `1` (which is required)
3. `execute_raw_command` is targeted specifically at controllers and does not support commands on channel 4

`pif_execute` really only services the EEPROM test, which (as you point out elsewhere) is more of a test for emulators than it is for libdragon.

I'll remove `pif_execute` and the EEPROM test from this PR. They were super helpful for confirming real hardware behavior and testing various emulators, but you're right that they're not really appropriate to be included in libdragon's tests.

Reply:
https://github.com/DragonMinded/libdragon/pull/125/commits/fd04a6359d93cd5f1d34906a97b287301ab69b55